content/usgs-packages/sbtools_Discovery.Rmd

```diff
@@ -0,0 +1,224 @@
+length(loc_results)
+head(loc_results, 2)
+```
+
+### Using `query_sb_doi`
+
+Use a Digital Object Identifier (DOI) to query ScienceBase. This should return only one list item, unless there is more than one ScienceBase item referencing this very unique identifier.
+
+```{r}
+# USGS Microplastics study 
+query_sb_doi('10.5066/F7ZC80ZP')
+
+####### I've tried A TON of DOIs I found through the web interface and just
+####### keep getting empty lists returned.
+query_sb_doi('10.1016/j.coldregions.2007.05.009')
+```
+
+### Using `query_sb_spatial`
+
+`query_sb_spatial` accepts 3 different methods for specifying a spatial area in which to look for data. You can specify a bounding box `bbox` as an `sp` spatial data object *[[[[NEED MORE TEXT HERE]]]]*.  Alternatively, you can supply a vector of latitudes and a vector of longitudes using `lat` and `long` arguments. The function will automatically use the minimum and maximum from those vectors to construct a boundary box. The last way to represent a spatial region to query ScienceBase is using a POLYGON Well-known text (WKT) object as a text string. The format is `"POLYGON(([LONG1 LAT1], [LONG2 LAT2], [LONG3 LAT3]))"`, where `LONG#` and `LAT#` are longitude and latitude pairs as decimals. See the [Open Geospatial Consortium WKT standard](http://www.opengeospatial.org/standards/wkt-crs) for more information. 
+
+```{r}
+### SPATIAL QUERY EXAMPLES NEED SOME TLC
+
+appalachia <- data.frame(
+  lat = c(34.576900, 36.114974, 37.374456, 35.919619, 39.206481),
+  long = c(-84.771119, -83.393990, -81.256731, -81.492395, -78.417345))
+
+conus <- data.frame(
+  lat = c(49.078148, 47.575022, 32.914614, 25.000481),
+  long = c(-124.722111, -67.996898, -118.270335, -80.125804))
+
+# verifying where points are supposed to be
+maps::map('usa')
+points(conus$long, conus$lat, col="red", pch=20)
+points(appalachia$long, appalachia$lat, col="green", pch=20)
+
+# query by bounding box
+bbox_sp_obj <- sp::SpatialPoints(appalachia)
+# query_sb_spatial(bbox=bbox_sp_obj)
+
+# query by latitude and longitude vectors
+query_sb_spatial(long = appalachia$long, lat = appalachia$lat)
+query_sb_spatial(long = conus$long, lat = conus$lat)
+
+# query by WKT polygon
+wkt_coord_str <- paste(conus$long, conus$lat, sep=" ", collapse = ",")
+wkt_str <- sprintf("POLYGON((%s))", wkt_coord_str)
+# query_sb_spatial(bb_wkt = wkt_str)
+```
+
+### Using `query_sb_date`
+
+`query_sb_date` returns ScienceBase items that fall within a certain time range. There are multiple timestamps applied to items, so you will need to specify which one to match the range. The default queries are to look for items last updated between 1970-01-01 and today's date. See `?query_sb_date` for more examples of which timestamps are available.
+
+```{r}
+# find data worked on in the last week
+today <- Sys.time()
+oneweekago <- today - (7*24*3600) # days * hrs/day * secs/hr
+recent_data <- query_sb_date(start = today, end = oneweekago)
```

Comment:
Tried doing this with Dates instead of datetimes, but still seeing this curious behavior...

```
length(query_sb_date(start = today, end = oneweekago, limit=10000))
[1] 10000
> length(query_sb_date(end = today, start = oneweekago, limit=10000))
[1] 407
> length(query_sb_date(start = as.Date(today), end = as.Date(oneweekago), limit=10000))
[1] 10000
> length(query_sb_date(end = as.Date(today), start = as.Date(oneweekago), limit=10000))
[1] 407
```